truss/tests/test_model_schema.py

```diff
@@ -0,0 +1,413 @@
+
+DEFAULT_CONFIG = """model_name: test-truss"""
+
+
+@pytest.mark.integration
+def test_truss_with_no_annotations():
+    truss_root = Path(__file__).parent.parent.parent.resolve()
+
+    truss_dir = truss_root / "truss" / "test_data" / "test_basic_truss"
+
+    tr = TrussHandle(truss_dir)
+
+    with ensure_kill_all():
+        _ = tr.docker_run(local_port=8090, detach=True, wait_for_server_ready=True)
+        truss_server_addr = "http://localhost:8090"
+        full_url = f"{truss_server_addr}/v1/models/model:predict"
+
+        response = requests.post(full_url, json={"prompt": "value"})
+        assert response.json() == {
+            "prompt": "value",
+        }
+
+        schema_response = requests.get(f"{truss_server_addr}/v1/models/model/schema")
+        assert schema_response.status_code == 404
+
+        schema_response.json()["error"] == "No schema found"
+
+
+@pytest.mark.integration
+def test_truss_with_annotated_inputs_outputs():
+    truss_root = Path(__file__).parent.parent.parent.resolve()
+
+    truss_dir = truss_root / "examples" / "annotated_types"
+
+    tr = TrussHandle(truss_dir)
+
+    with ensure_kill_all():
+        _ = tr.docker_run(local_port=8090, detach=True, wait_for_server_ready=True)
+        truss_server_addr = "http://localhost:8090"
+        full_url = f"{truss_server_addr}/v1/models/model:predict"
+
+        response = requests.post(full_url, json={"prompt": "value"})
+        assert response.json() == {
+            "generated_text": "value",
+        }
+
+        # An invalid input
+
+        response = requests.post(full_url, json={"bad_key": "value"})
+
+        assert response.status_code == 400
+        assert "error" in response.json()
+        assert "validation error" in response.json()["error"]
+
+        schema_response = requests.get(f"{truss_server_addr}/v1/models/model/schema")
+
+        schema = schema_response.json()
+
+        assert schema["input_schema"] == {
+            "properties": {"prompt": {"title": "Prompt", "type": "string"}},
+            "required": ["prompt"],
+            "title": "ModelInput",
+            "type": "object",
+        }
+
+        assert schema["output_schema"] == {
+            "properties": {
+                "generated_text": {"title": "Generated Text", "type": "string"}
+            },
+            "required": ["generated_text"],
+            "title": "ModelOutput",
+            "type": "object",
+        }
+        assert not schema["supports_streaming"]
+
+
+@pytest.mark.integration
+def test_truss_annotated_inputs_streaming_and_output():
+    streaming_truss = """
+from pydantic import BaseModel
+from typing import Union, Generator
+
+class ModelInput(BaseModel):
+    prompt: str
+
+class ModelOutput(BaseModel):
+    generated_text: str
+
+class Model:
+    def predict(self, request: ModelInput) -> Union[ModelOutput, Generator[str, None, None]]:
+        return ModelOutput(generated_text=request.prompt)
+"""
```

Comment:
Thoughts on making an `examples/annotated_types_streaming` truss so users have this as a guide for annotated streaming models? (I suppose this also applies to all the test cases below)

Reply:
I think we should honestly kill the examples and then focus on moving everything in the truss-examples repo over.

Reply:
You mean move the examples in this repo to truss-examples or the other way around?

Should we just move the trusses used for testing to some `test_data` subdirectory and remove all the examples not used for testing from examples?